### PR TITLE
Add docs for how to use a private registry with Kubernetes

### PR DESCRIPTION
**Pull request by iyovcheva**
This adds the following documentation: how to configure a private
registry (with link to Kubernetes documentation), how to create an
image pull secret and add the secret to an yaml file, how to pull
a function from a private Docker image and push it to a Docker
registry.

Signed-off-by: Ivana Yovcheva (VMware) <iyovcheva@vmware.com>

### CODE REVIEW

File: docs/deployment/kubernetes.md [hunk 1]
```diff
@@ -30,6 +30,115 @@ We have a special guide for minikube here:
...
+1. Update your `$HOME/.docker/config.json`:
+    * Run `docker login [server]` for each set of credentials you want to use
+    * Then check the `$HOME/.docker/config.json` file if the credentials are properly added
+2. Update your nodes:
+    * Get a list of your nodes:
+        * By name: `nodes=$(kubectl get nodes -o jsonpath='{range.items[*].metadata}{.name} {end}')`
+        * By IP: `nodes=$(kubectl get nodes -o jsonpath='{range .items[*].status.addresses[?(@.type=="ExternalIP")]}{.address} {end}')`
+    * Copy `$HOME/.docker/config.json` to the home directory of root on each node (f.e. `for n in $nodes; do scp ~/.docker/config.json root@$n:/root/.docker/config.json; done`)
+3. You can now test your cofiguration by creating a pod that uses a private image:
+
```
Comment (alexellis) [inline docs/deployment/kubernetes.md]: This section is not related to functions. Please could you omit it?

File: docs/deployment/kubernetes.md [hunk 2]
```diff
@@ -30,6 +30,92 @@ We have a special guide for minikube here:
...
+Log into the [Hub](https://hub.docker.com/) and make your image `private-alpine` private.
+
+Then create your openfaas project:
+
+```bash
+$ mkdir privatefuncs && cd privatefuncs
+$ touch stack.yaml
+```
+
+In your favoriate editor, open stack.yaml and add
```
Comment (alexellis) [inline docs/deployment/kubernetes.md]: Please could you run a spell-check on your work in future PRs? `favoriate` is a typo.

**Comment by alexellis** [comment]
This section is placed in the wrong place in the file. It interrupts the flow of why people come to the page - to deploy a cluster.

Please move the new section to the end of the document.

**Comment by iyovcheva** [comment]
@alexellis the comments are now addressed

File: docs/deployment/kubernetes.md [hunk 3]
```diff
@@ -185,6 +185,91 @@ $ echo -n "" | faas-cli invoke --gateway http://kubernetes-ip:31112 nodeinfo
...
+```bash
+export DOCKER_USERNAME=<your_docker_username>
+export DOCKER_PASSWORD=<your_docker_password>
+export DOCKER_EMAIL=<your_docker_email>
+```
+
+Then run this command to create the secret:
+
+```bash
+$ kubectl create secret docker-registry dockerhub \
```
Comment (alexellis) [inline docs/deployment/kubernetes.md]: @jmo808 @LucasRoesler is this still the correct secret type for the new Docker JSON config secret?

**Comment by alexellis** [comment]
Thank you for addressing the comments.

**Comment by mysim1** [comment]
Hi @alexellis 
My experience with AKS (Azure Kubernetes Services) is that when setting a private registry towards Azure Container Registries, those docker-registry secrets are not needed. Just setting az aks update --attach-acr $ACRID is enough to make refering images from <YOURNAME>.azurecr.io working.
